### PR TITLE
add rocksdb event metrics

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

File: src/util/rocksdb/event_listener.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#[derive(Default)]
+pub struct EventListener{}
+
+impl rocksdb::EventListener for EventListener {
+    fn on_flush_completed(&self, info: &FlushJobInfo) {
+        STORE_ENGINE_EVENT_COUNTER_VEC.with_label_values(&[info.cf_name(), "flush"]).inc();
+    }
+
+    fn on_compaction_completed(&self, info: &CompactionJobInfo) {
+        STORE_ENGINE_EVENT_COUNTER_VEC.with_label_values(&[info.cf_name(), "compaction"]).inc();
```
Comment (siddontang) [inline src/util/rocksdb/event_listener.rs]: can we know the compaction and ingestion, flush duration?

Comment (huachaohuang) [inline src/util/rocksdb/event_listener.rs]: We can't know the duration of flush and ingestion.
For compaction, we can get more stats from [here](https://github.com/facebook/rocksdb/blob/master/include/rocksdb/compaction_job_stats.h).
I think we can add `elasped_micros` and `num_corrupt_keys` here.

Comment (siddontang) [inline src/util/rocksdb/event_listener.rs]: yes, we can

File: src/util/rocksdb/engine_metrics.rs [hunk 2]
```diff
@@ -523,6 +523,20 @@ lazy_static!{
             &["type"]
         ).unwrap();
 
+    pub static ref STORE_ENGINE_COMPACTION_DURATIONS_VEC: HistogramVec =
+        register_histogram_vec!(
+            "tikv_engine_compaction_duration_seconds",
+            "Histogram of compaction duration seconds",
+            &["cf"]
```
Comment (siddontang) [inline src/util/rocksdb/engine_metrics.rs]: the default histogram bucket max time is only 10s, maybe too small, we should use our own buckets.

**Comment by siddontang** [comment]
PTAL @zhangjinpeng1987

**Pull request merged**